modules/apps/headless/headless-admin-content/headless-admin-content-test/src/testIntegration/java/com/liferay/headless/admin/content/resource/v1_0/test/StructuredContentResourceTest.java

```diff
@@ -85,6 +88,93 @@ public void setUp() throws Exception {
 		).build();
 	}
 
+	@Override
+	@Test
+	public void testGetSiteStructuredContentsPage() throws Exception {
+		super.testGetSiteStructuredContentsPage();
+
+		Locale locale = LocaleUtil.getDefault();
+
+		StructuredContent irrelevantStructuredContent =
+			_randomStructuredContent(locale);
+
+		com.liferay.headless.admin.content.client.resource.v1_0.
+			StructuredContentResource irrelevantStructuredContentResource =
+				_buildStructureContentResource(locale);
+
+		irrelevantStructuredContentResource.postSiteStructuredContentDraft(
+			testGetSiteStructuredContentsPage_getSiteId(),
+			irrelevantStructuredContent);
+
+		StructuredContent structuredContent = _randomStructuredContent(locale);
+
+		structuredContent.setPriority(Double.valueOf(1));
+
+		com.liferay.headless.admin.content.client.resource.v1_0.
+			StructuredContentResource structuredContentResource =
+				_buildStructureContentResource(locale);
+
+		StructuredContent postStructuredContent =
+			structuredContentResource.postSiteStructuredContentDraft(
+				testGetSiteStructuredContentsPage_getSiteId(),
+				structuredContent);
+
+		com.liferay.headless.delivery.client.dto.v1_0.StructuredContent
+			patchStructuredContent =
+				_structuredContentResource.patchStructuredContent(
+					postStructuredContent.getId(),
+					new com.liferay.headless.delivery.client.dto.v1_0.
+						StructuredContent() {
+
+						{
+							priority = Double.valueOf(3);
+						}
+					});
+
+		Page<StructuredContent> page =
+			structuredContentResource.getSiteStructuredContentsPage(
+				testGroup.getGroupId(), true, null, null, "priority eq 3.0",
+				Pagination.of(1, 10), null);
+
+		Assert.assertEquals(1, page.getTotalCount());
+
+		assertEqualsIgnoringOrder(
```

Comment:
Hi @Tim-Cao, here I think it would be better to get the first element of the list, and compare it directly to the patchStructuredContent (so to avoid converting to list). What do you think?

Reply:
Do you mean this?
```suggestion
assertEquals(_toStructuredContent(patchStructuredContent), page.fetchFirstItem());
```
I have a question, if compare the first item, how to make sure irrelevent item with other priority not shown in response?

Reply:
Sorry @Tim-Cao, you are 100% right, my bad

Reply:
Nvm